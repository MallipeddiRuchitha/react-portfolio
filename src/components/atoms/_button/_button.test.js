import React from 'react';
import { shallow ,mount} from 'enzyme';
import Button from './_button';
import '../../../setupTests'
describe("Button component",() =>{
    test("Matches the snapshot",() => {
        const button = shallow(<Button />);
        expect(button).toMatchSnapshot();

    })
    test("button element is rendered",() => {
        const wrapper = shallow(<Button />);
        expect(wrapper.find('#button').exists()).toEqual(true);
        
    });
    test("button element is enabled",() => {
        const wrapper =shallow(<Button value="save" disabled={false}/>);
        expect(wrapper.find("#button").props()['disabled']).toBe(false)
        
    });
    test("button element is disabled",() => {
        const wrapper =shallow(<Button value="save" disabled={true}/>);
        expect(wrapper.find("#button").props()['disabled']).toBe(true)
        
    });
   
    test("on click of button",() => {
        const mockOnClick = jest.fn();
        const wrapper =mount(<Button value="save" disabled={false} onClick={mockOnClick} />);
        //console.log(wrapper.debug());
        const button=wrapper.find("#button").last();
        button.simulate('click');
        wrapper.unmount();
        //fireEvent.click(getByRole("button"));
        expect(mockOnClick).toHaveBeenCalledTimes(1);
    });
});