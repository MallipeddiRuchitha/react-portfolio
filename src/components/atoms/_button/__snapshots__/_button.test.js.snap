// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Matches the snapshot 1`] = `
<div
  className="makeStyles-button-1"
>
  <WithStyles(ForwardRef(Button))
    color="primary"
    id="button"
    variant="contained"
  >
    <WithStyles(ForwardRef(Typography))
      variant="button"
    >
       
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Button))>
</div>
`;
